scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3852,452 @@ def rvs(self, dim, size=1, random_state=None):
+    are present in the population as in the example above. Consider
+    another example for a comparison with the hypergeometric distribution:
+
+    >>> from scipy.stats import hypergeom
+    >>> multivariate_hypergeom.pmf(x=[3, 1], m=[10, 5], n=4)
+    0.4395604395604395
+    >>> hypergeom.pmf(k=3, M=15, n=4, N=10)
+    0.43956043956044005
+
+    The functions ``pmf``, ``logpmf``, ``mean``, ``var``, ``cov``, and ``rvs``
+    support broadcasting, under the convention that the vector parameters
+    (``x``, ``m``, and ``n``) are interpreted as if each row along the last
+    axis is a single object. For instance, we can combine the previous two calls 
+    to `multivariate_hypergeom` as
+
+    >>> multivariate_hypergeom.pmf(x=[[8, 4], [3, 1]], m=[[10, 20], [10, 5]], n=[12, 4])
+    array([0.00252072, 0.43956044])
+
+    This broadcasting also works for ``cov``, where the output objects are
+    square matrices of size ``m.shape[-1]``. For example:
+
+    >>> multivariate_hypergeom.cov(m=[[7, 9], [10, 15]], n=[8, 12])
+    array([[[ 1.05, -1.05],
+            [-1.05,  1.05]],
+           [[ 1.56, -1.56],
+            [-1.56,  1.56]]])
+
+    That is, ``result[0]`` is equal to ``multivariate_hypergeom.cov(m=[7, 9], n=8)``
+    and ``result[1]`` is equal to ``multivariate_hypergeom.cov(m=[10, 15], n=12)``.
+
+    Alternatively, the object may be called (as a function) to fix the `m`
+    and `n` parameters, returning a "frozen" multivariate hypergeometric
+    random variable.
+
+    >>> rv = multivariate_hypergeom(m=[10, 20], n=12)
+    >>> rv.pmf(x=[8, 4])
+    0.0025207176631464523
+
+    See also
+    --------
+    scipy.stats.hypergeom : The hypergeometric distribution.
+    numpy.random.Generator.multivariate_hypergeometric : The multivariate hypergeometric distribution sampler in NumPy.
+    scipy.stats.multinomial : The multinomial distribution.
+
+    References
+    ----------
+    .. [1] The Multivariate Hypergeometric Distribution,
+           http://www.randomservices.org/random/urn/MultiHypergeometric.html
+
+    .. [2] Thomas J. Sargent and John Stachurski, 2020,
+           Multivariate Hypergeometric Distribution
+           https://python.quantecon.org/_downloads/pdf/multi_hyper.pdf
+    """  # noqa: E501
+    def __init__(self, seed=None):
+        super(multivariate_hypergeom_gen, self).__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__, mhg_docdict_params)
+
+    def __call__(self, m, n, seed=None):
+        """
+        Create a frozen multivariate_hypergeom distribution.
+
+        See `multivariate_hypergeom_frozen` for more information.
+        """
+        return multivariate_hypergeom_frozen(m, n, seed=seed)
+
+    def _process_parameters(self, m, n):
+        m = np.asarray(m, dtype=np.int_)
+        n = np.asarray(n, dtype=np.int_)
+        if m.ndim == 0:
+            raise ValueError("'m' must be an array!")
+
+        m, n = np.broadcast_arrays(m, n[..., np.newaxis])
+        n = n[..., 0]
```

Comment:
Have you thought about keeping `n` in its broadcasted shape here and `n` and `M` below? I suspect - although I admit, I haven't thought it all the way through - that it would reduce the need to do some of the funny things later on like `xcond_ = xcond_reduced | np.zeros(mncond.shape, dtype=np.bool_)`.
It appears to be working, but it's best to keep things tidy if we can so it can be maintained by others.

Reply:
The arrays broadcast twice. Once in `_process_parameters` and again in `_process_quatiles`. So, the broadcasted shape may change due to the array `x` in `_process_quatiles`. This makes it necessary to first align the conditions with the new broadcasted shape and then fill in the values. I can align the conditions (`xcond` and `mncond`) in `_process_quantiles` itself, but I think the code right now is cleaner than doing that.

Reply:
Though, I admit that that part of the code should ideally be in `_process_quantiles`.